app/src/main/java/com/example/weatherapp/models/utils/UnitSystemMapper.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.weatherapp.models.utils
+
+import com.example.weatherapp.models.measure.UnitSystem
+
+fun UnitSystem.mapTemperature(temperature: Double?): Double? {
+    return when (this.value) {
+        "metric" -> temperature?.fromKelvinToCelsius()
+        "imperial" -> temperature?.fromKelvinToFahrenheit()
```

Comment:
You don't need to use the value for the comparison here. You can use the enum name: `UnitSystem.Metric` for example